dijet/tasks/asymmetry.py

```diff
@@ -0,0 +1,176 @@
+        view.variance = np.apply_along_axis(np.cumsum, axis=axes_names.index("dijets_alpha"), arr=view.variance)
+
+        # Get integral of asymmetries as array
+        # Skip over-/underflow bins (i.e. TH1F -> ComputeIntegral for UHH2)
+        # h_all[{"dijets_asymmetry": sum}] includes such bins
+        integral = h_all.values().sum(axis=axes_names.index("dijets_asymmetry"), keepdims=True)
+
+        # Store for width extrapolation
+        h_nevts = h_all.copy()
+        h_nevts = h_nevts[{"dijets_asymmetry": sum}]
+        h_nevts.view().value = np.squeeze(integral)
+        self.output()["nevt"].dump(h_nevts, formatter="pickle")
+
+        # normalize histogram to integral over asymmetry
+        view.value = view.value / integral
+        view.variance = view.variance / integral**2
+
+        # Store asymmetries with gausstails for plotting
+        # TODO: h_all is further adjusted in scope of this task
+        #       retrospectivley changing this output as well?
+        self.output()["asym"].dump(h_all, formatter="pickle")
+
+        # Cut off non gaussian tails using quantiles
+        # NOTE: My first aim was to use np.quantile like
+        #       quantiles_lo = np.quantile(view.value, 0.015, axis=-1, keepdim=True)
+        #       This appraoch does not work with arrays containing histograms but only the raw data
+
+        # Get relative contribution from each bin to overall number.
+        # For each index, it adds the previous ones.
+        # For a normalized hist, the last entry must be 1.
+        percentage = np.cumsum(view.value, axis=-1)
+        # TODO: add assert so the last element is always 1
+
+        # Function to apply np.searchsorted on each 1D slice
+        def searchsorted_quantiles(cdf_slice, quantile):
+            return np.searchsorted(cdf_slice, quantile)
+
+        # TODO: As input parameter in task for uncertainties
+        #       Also needed for task output path
+        quantile_lo = 0.015  # 1.5 % / left tail
+        quantile_up = 0.985  # 98.5 % / right tail
+
+        # Find index of quantile
+        # NOTE: No alternative found yet for apply_along_axis
+        ind_lo = np.apply_along_axis(searchsorted_quantiles, -1, percentage, quantile_lo)
+        ind_up = np.apply_along_axis(searchsorted_quantiles, -1, percentage, quantile_up)
+
+        # Extend index array by one axis
+        ind_lo = np.expand_dims(ind_lo, axis=-1)
+        ind_up = np.expand_dims(ind_up, axis=-1)
+
+        # Obtain bin edge value for qunatile index
+        # Store in histogram structure
+        # NOTE: Not sure to rebin a hsitogram from (:,:,80) to (:,:,1)
+        #       Remove bin completly with sum, since only one value is needed
+        asym_edges = h_all.axes["dijets_asymmetry"].edges  # One dim more then view.value
+        asym_edges_lo = asym_edges[ind_lo]  # Get value for lower quantile
+        asym_edges_up = asym_edges[ind_up]  # Store in histogram structure
```

Comment:
For the upper quantile, it might be better to use the upper edge of the bin returned by `np.searchsorted` to ensure that the cutoff happens symmetrically for the upper and lower tails. To achieve this, the upper index can be incremented by one, as suggested below.

```suggestion
        asym_edges_up = asym_edges[ind_up + 1]  # Store in histogram structure
```